src/GitVersionCore/GitPreparer.cs

```diff
@@ -150,6 +150,21 @@ public string GetDotGitDirectory()
             return dotGitDirectory;
         }
 
+        public string GetDotGitDirectoryOfMain()
+        {
+            var dotGitDir = GetDotGitDirectoryOfWorktreeOrMain();
+
+            if (!dotGitDir.Contains(Path.Combine(".git", "worktrees")))
+                return dotGitDir;
+
+            return Directory.GetParent(Directory.GetParent(dotGitDir).FullName).FullName;
+        }
+
+        public string GetDotGitDirectory()
+        {
+            return GetDotGitDirectoryOfMain();
```

Comment:
A method just invoking another method feels a bit unnecessary?

Reply:
I agree. I tried to minimize the changes to existing code, but it would be clearer if we'd replace `GetDotGitDirectory` with `GetDotGitDirectoryOfMain` throughout the code base.

Reply:
I'm not to fond of these names either. Maybe we can come up with something better.

_Main working tree_ and _linked working tree_ is what is used in the git documentation: "A repository has one main working tree (if it’s not a bare repository) and zero or more linked working trees." (https://git-scm.com/docs/git-worktree).

We could use `GetDotGitDirectoryOfRepo` (instead of `GetDotGitDirectoryOfMain`) and `GetDotGitDirectoryOfWorktree` (instead of `GetDotGitDirectoryOfWorktreeOrMain`), although the latter method name doesn't make it obvious that it returns `.git` directory of the main or linked working tree.

Reply:
That would be an improvement, yes. 👍

Reply:
Done.